libbeat/processors/communityid/communityid.go

```diff
@@ -211,43 +213,31 @@ func tryToIP(from interface{}) (net.IP, bool) {
 	}
 }
 
-// tryToUint16 tries to coerce the given interface to an uint16. On success it
+// tryToUint tries to coerce the given interface to an uint16. On success it
 // returns the int value and true.
-func tryToUint16(from interface{}) (uint16, bool) {
+func tryToUint(from interface{}) (uint, bool) {
 	switch v := from.(type) {
 	case int:
-		return uint16(v), true
+		return uint(v), true
 	case int8:
-		return uint16(v), true
-	case int16:
-		return uint16(v), true
-	case int32:
-		return uint16(v), true
-	case int64:
-		return uint16(v), true
+		return uint(v), true
```

Comment:
```suggestion
		return uint(v), true
	case int16:
		return uint(v), true
	case int32:
		return uint(v), true
	case int64:
		return uint(v), true
```

Reply:
I'll try this but the weird thing is I created a test case to communityid_test.go, with the exact same parameters as the events that lost their community id and the test outputted the expected string. Also I did some other testing to check which case in the switch was being hit and it seemed no matter what number I put, it was always the int.